### PR TITLE
Fix/cmake find mode

### LINKED ISSUE #10930: Check find_mode potential issue before 1.47
Hi @lasote! Sorry, have seen PR only now (when it has already merged :)).

Here should be not `self.find_module_mode`, but something which depends on `values[find_mode.lower()]` from 191 line. Like `values[find_mode.values()] == "MODULE"` (or so) (doesn't matter what will be here for "none" find_mode). This is important for dependencies with different `cmake_file_name` and `cmake_module_file_name`.

UPD: Not only here, but in `_get_dependency_filenames` too.

_Originally posted by @alex-700 in https://github.com/conan-io/conan/pull/10917#discussion_r838721838_

### PR DESCRIPTION
**Pull request by lasote**
Changelog: omit
Docs: omit

Close #10930

@alex-700 How does it look?

### CODE REVIEW

File: conan/tools/cmake/cmakedeps/templates/target_data.py [hunk 1]
```diff
@@ -181,13 +181,12 @@ def _get_dependencies_find_modes(self):
...
-            dep_file_name = get_file_name(dep, self.find_module_mode)
-            find_mode = dep.cpp_info.get_property("cmake_find_mode")
+            dep_file_name = get_file_name(dep)
+            find_mode = get_find_mode(dep)
             values = {"none": "",
                       "config": "NO_MODULE",
                       "module": "MODULE",
                       "both": "NO_MODULE" if not self.find_module_mode else "MODULE"}
-            find_mode = find_mode or "both"
             ret[dep_file_name] = values[find_mode.lower()]
```
Comment (alex-700) [inline conan/tools/cmake/cmakedeps/templates/target_data.py]: nit:
```suggestion
            ret[dep_file_name] = values[find_mode]
```

File: conan/tools/cmake/utils.py [hunk 2]
```diff
@@ -4,14 +4,22 @@ def is_multi_configuration(generator):
...
     ret = conanfile.cpp_info.get_property("cmake_file_name")
     return ret or conanfile.ref.name
 
+
+def get_find_mode(conanfile):
+    """
+    :param conanfile: conanfile of the requirement
+    :return: "none" or "config" or "module" or "both"
+    """
+    tmp = conanfile.cpp_info.get_property("cmake_find_mode") or "both"
```
Comment (alex-700) [inline conan/tools/cmake/utils.py]: Documentation [here](https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmakedeps.html) says, that `cmake_find_mode` defaulted to "config". So should it be `or "config"`?

File: conan/tools/cmake/utils.py [hunk 3]
```diff
@@ -4,14 +4,22 @@ def is_multi_configuration(generator):
     return "Visual" in generator or "Xcode" in generator or "Multi-Config" in generator
 
 
-def get_file_name(conanfile, find_module_mode=False):
+def get_file_name(conanfile, forced_module_mode=None):
     """Get the name of the file for the find_package(XXX)"""
     # This is used by the CMakeToolchain to adjust the XXX_DIR variables and the CMakeDeps. Both
     # to know the file name that will have the XXX-config.cmake files.
-    if find_module_mode:
+    if forced_module_mode or get_find_mode(conanfile) == "module":
```
Comment (alex-700) [inline conan/tools/cmake/utils.py]: Should `FIND_MODE_MODULE` constant from [here](https://github.com/conan-io/conan/blob/develop/conan/tools/cmake/cmakedeps/cmakedeps.py#L14) be used? Should other hardcoded strings in this file be fixed too?

**Review by alex-700 — CHANGES_REQUESTED** [review]

File: conan/tools/cmake/utils.py (hunk 2 above)
Comment (lasote) [reply conan/tools/cmake/utils.py]: Mhh not in this case, but yes, this is a bit messy so I will return None when not set.

File: conan/tools/cmake/utils.py [hunk 4]
```diff
@@ -21,5 +24,7 @@ def get_find_mode(conanfile):
     :param conanfile: conanfile of the requirement
     :return: "none" or "config" or "module" or "both"
```
Comment (alex-700) [inline conan/tools/cmake/utils.py]: doccomment should be updated for new return value None.

**Review by alex-700 — APPROVED** [review]
LGTM!
Together with the previous PR, this is a great fix!

File: conan/tools/cmake/utils.py (hunk 2 above)
Comment (alex-700) [reply conan/tools/cmake/utils.py]: Thanks for including comment in the "convertation to `{{ _DEPENDENCY }} _FIND_MODE`": now I understand this!

**Review by memsharded — APPROVED** [review]

**Pull request merged**